src/main/PersistentState.cpp

```diff
@@ -120,6 +207,64 @@ PersistentState::setRebuildForType(LedgerEntryType let)
     updateDb(getStoreStateName(kRebuildLedger, let), "1");
 }
 
+void
+PersistentState::upgradeSCPDataV1Format()
+{
+    for (uint32_t i = 0; i <= mApp.getConfig().MAX_SLOTS_TO_REMEMBER; i++)
+    {
+        std::string oldStateName = getStoreStateName(kLastSCPDataXDR, i);
+        auto val = getFromDb(oldStateName);
+        if (val.empty())
+        {
+            continue;
+        }
+        std::vector<uint8_t> buffer;
+        decoder::decode_b64(val, buffer);
+
+        PersistedSCPState scpState;
+
+        try
+        {
+            xdr::xdr_from_opaque(buffer, scpState);
+            if (scpState.v() != 0)
+            {
+                throw std::runtime_error("Invalid persisted state version");
+            }
+
+            PersistedSCPState newScpState;
+            newScpState.v(1);
+            newScpState.v1().scpEnvelopes = scpState.v0().scpEnvelopes;
+            newScpState.v1().quorumSets = scpState.v0().quorumSets;
+
+            std::unordered_map<Hash, std::string> txSets;
+            for (auto const& txSet : scpState.v0().txSets)
+            {
+                auto txSetPtr = TxSetFrame::getTxSetFramePtr(txSet, mApp);
+                txSets.emplace(
+                    std::pair(txSetPtr->getContentsHash(),
```

Comment:
nit: there is no need to use the pair constructor when you're using `emplace`

Reply:
True, just a habit of writing things down explicitly. Updated (in the other place too)